velox/vector/tests/VectorTest.cpp

```diff
@@ -1671,3 +1671,64 @@ TEST_F(VectorTest, multipleDictionariesOverLazy) {
     ASSERT_EQ(i, dict->as<SimpleVector<int32_t>>()->valueAt(i));
   }
 }
+
+class LazyTestingLoader : public VectorLoader {
+ public:
+  explicit LazyTestingLoader(RowVectorPtr data, int32_t* rowCounter)
+      : data_(data) {
+    rowCounter_ = rowCounter;
+  }
+
+  void loadInternal(RowSet rows, ValueHook* hook, VectorPtr* result) override {
+    (*rowCounter_) += rows.size();
+    *result = data_;
+  }
+
+ private:
+  RowVectorPtr data_;
+  int32_t* rowCounter_;
+};
+
+/// Test lazy loading of dictionary vector
+TEST_F(VectorTest, selectiveLoadingOfLazyDictionary) {
+  auto vectorMaker = std::make_unique<test::VectorMaker>(pool_.get());
+
+  vector_size_t size = 10;
+  auto indices =
+      makeIndices(size, [&](auto row) { return (row % 2 == 0) ? row : size; });
+  int32_t rowCounter = 0;
+  auto rowData = makeRowVector(
```

Comment:
Any particular reason to use a RowVector here? A flat integer vector work work as well.

Reply:
Yea I was initially using a flat int vector but then during loading in LazyVector the actual vector was not created due to this check:

    if (!vector_ && type_->kind() == TypeKind::ROW) {
      vector_ = BaseVector::create(type_, rows.back() + 1, pool_);
    }
    
Any particular reason why the vector_ is only created for row types?

Reply:
@patrickstuedi Where can I see the code you are pasting here?

Reply:
Got it. You don't need the 'result' vector to be initialized. `*result = data_;` this code works regardless of whether result is initialized or not, right?

Reply:
You're right, not sure anymore why I thought it was an issue.